src/test/groovy/PublishUrlTest.groovy

```diff
@@ -58,6 +58,18 @@ class PublishUrlTest extends Specification {
         getPublishUrlRunner().buildAndFail()
     }
 
+    @Unroll
+    def "Publish URL is empty for HEAD branch on build server: #HEAD"() {
```

Comment:
You don't need @Unroll and #HEAD here. That's only for parameterized tests (i.e. tests that have a where: section at the end).